cocos/2d/CCFontAtlasCache.cpp

```diff
@@ -88,22 +88,53 @@ FontAtlas* FontAtlasCache::getFontAtlasTTF(const _ttfConfig* config)
     return nullptr;
 }
 
-FontAtlas* FontAtlasCache::getFontAtlasFNT(const std::string& fontFileName, const Vec2& imageOffset /* = Vec2::ZERO */)
+FontAtlas* FontAtlasCache::getFontAtlasFNT(const std::string& fontFileName)
 {
-    auto realFontFilename = FileUtils::getInstance()->getNewFilename(fontFileName);  // resolves real file path, to prevent storing multiple atlases for the same file.
+    return getFontAtlasFNT(fontFileName, Rect::ZERO, false);
+}
+
+FontAtlas* FontAtlasCache::getFontAtlasFNT(const std::string& fontFileName, const std::string& subTextureKey)
+{
+    const auto realFontFilename = FileUtils::getInstance()->getNewFilename(fontFileName);  // resolves real file path, to prevent storing multiple atlases for the same file.
+    std::string atlasName = subTextureKey + " " + realFontFilename;
+
+    const auto it = _atlasMap.find(atlasName);
+    if (it == _atlasMap.end())
+    {
+        const auto font = FontFNT::create(realFontFilename, subTextureKey);
+
+        if (font)
+        {
+            const auto tempAtlas = font->createFontAtlas();
+            if (tempAtlas)
+            {
+                _atlasMap[atlasName] = tempAtlas;
+                return _atlasMap[atlasName];
+            }
+        }
+    }
+    else
+        return it->second;
+
+    return nullptr;
+}
+
+FontAtlas* FontAtlasCache::getFontAtlasFNT(const std::string& fontFileName, const Rect& imageRect, bool imageRotated)
+{
+    const auto realFontFilename = FileUtils::getInstance()->getNewFilename(fontFileName);  // resolves real file path, to prevent storing multiple atlases for the same file.
```

Comment:
I think should use FileUtils::getInstance()->fullPathForFilename instead.

Reply:
Same for this, I never modified this code, that's all from the current implementation. I can change it if you require, but are you certain it would work?